src/please.go

```diff
@@ -864,6 +867,61 @@ var buildFunctions = map[string]func() int{
 	},
 }
 
+// Check if tool is given as label or path and then run
+func CheckIsLabelOrPathAndRun(config *core.Configuration, _tool tool.Tool, args []string) {
+//	log.Warning("Sam: Hi I'm in CheckIsLabelOrPathAndRun()")
+	tools := tool.MatchingTools(config, string(_tool))
+//	log.Warning("Sam: tools=%s", tools)
+	target := fs.ExpandHomePath(tools[tool.AllToolNames(config, string(_tool))[0]])
+	label := core.ParseBuildLabels([]string{target})
+//	log.Warning("Sam: target=%s", target)
+	if !core.LooksLikeABuildLabel(target) {
+//		log.Warning("Sam: Code path 1")
+		if !filepath.IsAbs(target) {
+			t, err := core.LookBuildPath(target, config)
+			if err != nil {
+				log.Fatalf("%s", err)
+			}
+			target = t
+		}
+		// Hopefully we have an absolute path now, so let's run it.
+		if success, state := runBuild(label, true, false, false); success {
+			var dir string
+			if opts.Run.InWD {
+//				log.Warning("Sam: Changed dir to originalWorkingDirectory")
+				dir = originalWorkingDirectory
+			}
+
+			annotatedOutputLabels := core.AnnotateLabels(label)
+			//			annotatedOutputLabels := state.ExpandOriginalMaybeAnnotatedLabels([]core.AnnotatedOutputLabel{opts.Tool.Args.Tool})
+			if len(annotatedOutputLabels) != 1 {
+				log.Fatalf("%v expanded to more than one target. If you want to run multiple targets, use `plz run parallel %v` or `plz run sequential %v`. ", opts.Run.Args.Target, opts.Run.Args.Target, opts.Run.Args.Target)
+			}
+
+			run.Run(state, annotatedOutputLabels[0], opts.Tool.Args.Args.AsStrings(), opts.Run.Remote, opts.Run.Env, opts.Run.InTempDir, dir, opts.Run.Cmd)
+		}
+		//		err := syscall.Exec(target, append([]string{target}, args...), os.Environ())
+		//		log.Fatalf("Failed to exec %s: %s", target, err) // Always a failure, exec never returns.
+	}
+//	log.Warning("Sam: Code path 2")
+	// The tool is allowed to be an in-repo target. In that case it's essentially equivalent to "plz run".
+	// We have to re-exec ourselves in such a case since we don't know enough about it to run it now.
+	//	plz, _ := os.Executable()
+	//	args = append([]string{os.Args[0], "run", target, "--"}, args...)
+	//Run()
+	//	err := syscall.Exec(plz, args, os.Environ())
+	//	log.Fatalf("Failed to exec %s run %s: %s", plz, target, err) // Always a failure, exec never returns.
+
+	if success, state := runBuild(label, true, false, false); success {
+		var dir string
+		if opts.Run.InWD {
+			dir = originalWorkingDirectory
+		}
+		annotatedOutputLabels := core.AnnotateLabels(label)
+		run.Run(state, annotatedOutputLabels[0], opts.Tool.Args.Args.AsStrings(), opts.Run.Remote, opts.Run.Env, opts.Run.InTempDir, dir, opts.Run.Cmd)
```

Comment:
We want to pass options from the run sub-command. They won't be populated because we're running the `tool` sub-command here. That's not a problem as we don't actually want to run tools like that. This can probably just be something like:
```
run.Run(state, annotatedOutputLabels[0], opts.Tool.Args.Args.AsStrings(), false,  nil, false, "", "")
```